pkg/openthread/contrib/openthread.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2017 Fundacion Inria Chile
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @ingroup     net
+ * @file
+ * @brief       Implementation of OpenThread main functions
+ *
+ * @author      Jose Ignacio Alamos <jialamos@uc.cl>
+ * @}
+ */
+
+#include <assert.h>
+
+#include "openthread/platform/alarm.h"
+#include "openthread/platform/uart.h"
+#include "ot.h"
+#include "random.h"
+#include "thread.h"
+#include "xtimer.h"
+
+#ifdef MODULE_AT86RF2XX
+#include "at86rf2xx.h"
+#include "at86rf2xx_params.h"
+#endif
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifdef MODULE_AT86RF2XX     /* is mutual exclusive with above ifdef */
+#define OPENTHREAD_NETIF_NUMOF        (sizeof(at86rf2xx_params) / sizeof(at86rf2xx_params[0]))
+#endif
+
+#ifdef MODULE_AT86RF2XX
+static at86rf2xx_t at86rf2xx_dev;
+#endif
+
+#define OPENTHREAD_NETDEV_BUFLEN (ETHERNET_MAX_LEN)
+
+static uint8_t rx_buf[OPENTHREAD_NETDEV_BUFLEN];
+static uint8_t tx_buf[OPENTHREAD_NETDEV_BUFLEN];
+static char ot_thread_stack[2 * THREAD_STACKSIZE_MAIN];
+
+/* init and run OpeanThread's UART simulation (stdio) */
+void openthread_uart_run(void)
+{
+    char buf[256];
+    char c;
+    msg_t msg;
+
+    msg.type = OPENTHREAD_SERIAL_MSG_TYPE_EVENT;
+    msg.content.ptr = buf;
+
+    buf[1] = 0;
+    while (1) {
+        c = getchar();
+        buf[0] = c;
+        msg_send(&msg, openthread_get_pid());
+    }
+}
+
+void openthread_bootstrap(void)
```

Comment:
this function is called for auto init. It would be great to activate the interface (e.g ifconfig up) at the end of it, set the panid, channel and start thread at the end of the function.
The default params could be defined in a `openthread_params.h` file available in the test directory (or somewhere else better).
What do you think ?